aea/crypto/ledger_apis.py

```diff
@@ -238,13 +238,13 @@ def generate_tx_nonce(
         :param client: the address of the client.
         :return: return the hash in hex.
         """
-        assert identifier in self.apis.keys()
```

Comment:
I don't like the solution. Can we instead have an if else block?
`if identifier in self.apis.keys():
     try:
             api = self.apis[identifier]
             tx_nonce = api.generate_tx_nonce(seller=seller, client=client)		             tx_nonce = api.generate_tx_nonce(seller=seller, client=client)
         except Exception:
             logger.warning(
                 "An error occurred while attempting to generate the tx_nonce"
             )
 else:
    logger.warning(" You didn't specify a ledger so the tx_nonce will be Empty.")
    tx_nonce= ""
return tx_nonce
`